### ISSUE TITLE
FP16 model export doesn't work on Jetson

### ISSUE THREAD

**Issue by Less6990** [ultralytics/ultralytics#21762]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

I am trying to export a YOLO11n.pt model with FP16 precision (in tensorRT format), on which we fine-tuned with a proprietary dataset, using the "half=True" flag as per the documentation, however during the execution of a proprietary script, by logging the precision of the model, I obtain the result "half=false", therefore I am actually not exporting the model with 16-bit precision.

I am using the autobackend class with the aim of eliminating pre-processing and post-processing since I manage it independently, and also to obtain a tensor as output, avoiding creating overhead in processing the result obtained on CPU instead using YOLO class.  These choices are due to the implementation of a real time computer vision system.  The export function is the one used in the documentation (as shown below)

I carry out the test on an nvidia jetson orin agx 64 GB, which allows us to obtain an inference time of 6.18 ms, while by carrying out the same test on my laptop which has an nvidia rtx 4070, I obtain an inference time of approximately 3 ms, and also by logging precision on the laptop, I get half=true, so it actually seems that on my laptop I can get, with the same code, a model with 16-bit precision.

### Environment

I am operating on a nvidia jetson agx orin 64GB

numpy ✅= 1.23.5
torch ✅= 2.7.0
torchvision ✅= 0.22.0
ultralytics ✅= 8.3.117
tensorrt ✅= 10.3.0

### Minimal Reproducible Example

# Using backend class

self.backend = AutoBackend(
            weights=weights,
            device=select_device(device),
            # data=self.args.data,
            fp16=True,
            fuse=True,
            verbose=verbose,
        )

self.half = self.backend.fp16

# Ultralytics export function

from ultralytics import YOLO

model = YOLO("best.pt")

model.export(
    format="engine",
    dynamic=False,
    batch=4,
    workspace=2,
    half=True,
    imgsz=640,
    device=0,
)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Less6990, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your report and assist you soon.

We recommend a visit to the Docs for new users where you can find many Python and CLI usage examples and where many of the most common questions may already be answered:
- Docs: https://docs.ultralytics.com/
- Python: https://docs.ultralytics.com/usage/python/
- CLI: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it: https://docs.ultralytics.com/help/minimum-reproducible-example/

Since this appears related to Export and FP16/Jetson behavior, please also include:
- Full export console logs showing TensorRT builder/config details (including any FP16 layers/flags reported)
- The exact AutoBackend initialization and inference snippet you’re using on Jetson, plus how you’re “logging the precision” (code/output)
- Outputs of your environment on Jetson: Python, CUDA/JetPack, NVIDIA driver, TensorRT build, and `ultralytics` version
- Whether the same `.engine` file is used across devices or rebuilt per device
- Results of running the same export with a single image size and batch=1 for a quick check
- Any warnings during export or first inference load on Jetson
These details will help us reproduce and diagnose the FP16 detection discrepancy between your Jetson AGX Orin and RTX 4070 setups 🙏

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best. For real-time chat, head to Discord 🎧: https://discord.com/invite/ultralytics. Prefer in-depth discussions? Check out Discourse: https://community.ultralytics.com/. Or dive into threads on our Subreddit to share knowledge with the community: https://www.reddit.com/r/Ultralytics/

## Upgrade
Upgrade to the latest `ultralytics` package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:
- Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml
- Python: https://www.python.org/
- PyTorch: https://pytorch.org/get-started/locally/
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit:
- Modes: https://docs.ultralytics.com/modes/
- Tasks: https://docs.ultralytics.com/tasks/

**Comment by nikunjlad** [comment]
@Less6990 For the Jetson ecosystems, I have always had success exporting a quantized model using the Deepstream plugins of Nvidia. I would suggest the below since that's how I got the FP16/INT8 models out.

1. Take a model trained by ultralytics (which would usually be exported in .pt format). 
2. Convert a trained .pt model to onnx (this is to ensure that it is in a portable format for any platform). You can either do this conversion on the x86 system or on the target hardware where you want to quantize your model. Make sure to create a new virtualenv with the necessary libraries for this conversion. 
3. Using the .onnx model, use the Deepstream YOLO repo and follow the instructions to quantize YOLOv11.pt . https://github.com/marcoslucianops/DeepStream-Yolo/blob/master/docs/YOLO11.md
4. Depending on which version of Jetpack you are using, make sure to install the correct Deepstream version. Deepstream is a framework provided by Nvidia which gives access to hardware accelerated Gstreamer plugins. They help to get a quantized model out with FP16/INT8 precision. (Make use of the sample configs under /opt/nvidia/deepstream/deepstream folder to get export of your model)

So far with the above 4 step process, I have quantized yolov7 and yolov9 models in INT8/FP16 precision on Jetson Orin NX running Jetpack 6.0 and Jetson Xavier AGX running Jetpack 4.6

**Comment by Y-T-G** [comment]
`fp16` would be `True` only if the TensorRT input expects FP16. But this doesn't tell you whether the engine is actually FP16. TensorRT engines can take in FP32 input even if you export with FP16.

https://github.com/ultralytics/ultralytics/issues/21264#issuecomment-3031307482

The correct way to check if it exported to FP16 is to check the file size. The file size of FP16 would be close to the `.pt` file (which is also in half precision) and for FP32, it would be close to double the size.
